### PR TITLE
Clarify install instructions

### LINKED ISSUE #340: Suggestion: include warnings about stale cached files during the installation process
Until recently, I had successfully been able to build the development releases using [your instructions](https://github.com/ipython/ipywidgets/blob/6836a4777050316cdfe7ffa378d47b72f511d95a/README.md#development-install):

```
git clone https://github.com/ipython/ipywidgets
cd ipywidgets
pip install -v -e .
```

But then I started to getting complaints in the browser console that `ipywidgets` wasn't properly installed.  After doing more experimentation than I was hoping, I finally realized `pip install -v -e .` was making use of cached results from previous builds.  Once I cleaned out files and directories not directly under git control with

```
git clean -d -f -x
```
I was able to install `ipywidgets`.

Is cleaning out cached files during installations something others have had to do?

### PR DESCRIPTION
**Pull request by jdfreder**
ping @SylvainCorlay - do I have the version compatibility table right?

ping @minrk or @willingc - doc/tech content review if you get a chance... "Does this make sense?"

closes #340

### CODE REVIEW

**Comment by jdfreder** [comment]
Motivated by install troubles expressed by users in #377

File: README.md [hunk 1]
```diff
@@ -28,9 +36,15 @@ If you're interested in only installing the JavaScript, you may do so by running
...
 
-To install ipywidgets from git, you will need [npm](https://www.npmjs.com/).  If
-you install using sudo, you need to make sure that npm is also available in the
-PATH used with sudo.
+To install ipywidgets from git, you will need [npm](https://www.npmjs.com/) and
+the latest [development copy of the Jupyter
+notebook](https://github.com/jupyter/notebook) because everything in the
+ipywidgets repository is developed using Jupyter notebook master. If you want
+to have a copy of ipywidgets that works against a stable version of the
+notebook, checkout the appropriate tag (see the compatibility table above).
```
Comment (willingc) [inline README.md]: I would capitalize and bold "Compatibility"

File: README.md [hunk 2]
```diff
@@ -41,16 +55,33 @@ PATH used with sudo.
...
+3. Install the Jupyter Widgets nbextension
 
         cd widgetsnbextension
+        npm install
+        npm run update:widgets
         pip install -v -e .
+        cd ..
 
-3. When you've made changes to `jupyter-js-widgets` and you'd like to test those
+Later, when you've made changes to `jupyter-js-widgets` and you'd like to test those
```
Comment (willingc) [inline README.md]: After ~~Later, when~~

File: README.md [hunk 3]
```diff
@@ -41,16 +55,33 @@ PATH used with sudo.
...
 changes, run the following, empty your browsers cache, and refresh the page.
 
         cd widgetsnbextension
         npm run update:widgets
+        cd ..
+
+TIPS: If you have any problems with the above install procedure, make sure than
```
Comment (willingc) [inline README.md]: than/t

**Comment by willingc** [comment]
@jdfreder A few minor edits. Also any reason not to use ``git clean -dfx`` in line 81? Otherwise looks good.

**Comment by jdfreder** [comment]
> @jdfreder A few minor edits. Also any reason not to use git clean -dfx in line 81? Otherwise looks good.

Nope, just me being lazy- I copied and pasted the line from #340.  I'll change it, thanks!

**Comment by jdfreder** [comment]
:heavy_check_mark: @willingc 's review comments

File: README.md (hunk 1 above)
Comment (jdfreder) [inline README.md]: :heavy_check_mark:

File: README.md (hunk 2 above)
Comment (jdfreder) [inline README.md]: :heavy_check_mark:

File: README.md (hunk 3 above)
Comment (jdfreder) [inline README.md]: :heavy_check_mark:

**Comment by SylvainCorlay** [comment]
:+1:

**Comment by minrk** [comment]
Thanks!

**Pull request merged**